code-challenges.test.js

```diff
@@ -14,39 +14,108 @@
+describe("remainOnly", () => {
+  const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+  // Expected output: [ 2, 0, -1, 0 ]
+  const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+  // Expected output: [ 2, 1, -1 ]
+  it("Returns an array of only the REMAINDERS of the number when divided by 3", () => {
+    expect(remainOnly(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(remainOnly(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+//GOOD FAILURE ReferenceError: remainOnly is not defined
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Input: Array
+//Output: Array of numbers
+//Process: Create a function. The function takes in an array. Filter through the array to return a new array of the typeof values that are numbers. Then iterate through the new array using .map to find the remainder of each value.
+
+const remainOnly = (arr) => {
+  const findData = arr.filter((value) => typeof value === 'number') //Filters through given array and looks for the typeof data type that is a 'number'
+  return findData.map((value) => { //Iterate through the new array. 
+    return value % 3 //Finds the remainder of each value divisible by three
+  })
+}
+// console.log(remainOnly(hodgepodge1))
+// console.log(remainOnly(hodgepodge2))
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("sumCubed", () => {
+  const cubeAndSum1 = [2, 3, 4]
+  // Expected output: 99
+  const cubeAndSum2 = [0, 5, 10]
+  // Expected output: 1125
+  it("Returns the sum of the numbers cubed", () => {
+    expect(sumCubed(cubeAndSum1)).toEqual(99)
+    expect(sumCubed(cubeAndSum2)).toEqual(1125)
+  })
+})
+//GOOD FAILURE ReferenceError: sumCubed is not defined
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//Input: Array
+//Output: Number
+//Process: Create a function that takes in an array. Create two new variables, one that holds an empty array and another that holds a number. Then create a for loop to iterate through the array that is being passed through the function. Then cube each number in the array. Push those numbers, since now they aren't in an array, into an empty array. Then add each of the numbers in the new array to get the sum and return the new sum.
+
+const sumCubed = (arr) => {
+  let newArr = [] //A variable that holds an empty array
+  let sum = 0 //A variable that holds a number
+  for (let i = 0; i < arr.length; i++) { //Iterates through array being passed
+    let cubed = arr[i] * arr[i] * arr[i] //cubed stores the arr indexes being cubed
+    newArr.push(cubed) //Pushes each number into empty array
+    sum = newArr[0] + newArr[1] + newArr[2] // Adds each number in the array, now that it is cubed. 
```

Comment:
Remember to keep your code dynamic. If we were to pass an array that has 100 numbers in it for example, the function would not get the proper sum at the end.